rasa/shared/core/events.py

```diff
@@ -503,11 +503,13 @@ def __eq__(self, other: Any) -> bool:
         return (
             self.text,
             self.intent_name,
-            [jsonpickle.encode(ent) for ent in self.entities],
+            [
+                jsonpickle.encode(sorted(ent)) for ent in self.entities
```

Comment:
regex message handler was creating things in a different order which means events weren't matching...

Reply:
I can't remember - but essentially it was just because the ordering of the entities was different.